### DISCUSSION TITLE
QuadPoints for re-use in the Adobe Embed API

### DISCUSSION THREAD

**Discussion by erswelljustin** [pymupdf/PyMuPDF#2309] [Looking for help]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
We are trying to extract annotations from a PDF and then use that data to 'Cherry Pick' the annotations we require to display them in a clean version of the PDF using the Adobe Embed API. We are getting data fine from the PDF using PyMuPDF however when we apply the annotation to the clean PDF we are getting weird results.

## To Reproduce (mandatory)
First parse the Annotated PDF (
[Sandborn2003Annotated.pdf](https://github.com/pymupdf/PyMuPDF/files/11104205/Sandborn2003Annotated.pdf)
)
Then using the Adobe Document Cloud Embed API - https://developer.adobe.com/document-services/docs/overview/pdf-embed-api/ apply the first Annotation in the stored json to a clean version of the file (
[Sandborn2003.pdf](https://github.com/pymupdf/PyMuPDF/files/11104216/Sandborn2003.pdf)
)

## Expected behavior (optional)
We expect to see the clean PDF take on the same annotation as the source

## Screenshots (optional)
Source Annotated Document (
<img width="664" alt="Screenshot 2023-03-29 at 19 25 45" src="https://user-images.githubusercontent.com/1675504/228633129-9096db5a-b5e9-4640-984b-5de60db4087a.png">
)

Target PDF after applying the annotation (
![screenshot_2023-03-29_at_20 59 45_720](https://user-images.githubusercontent.com/1675504/228633400-2db8cd6d-f57d-4982-a3a6-f3cc12b507bc.png))

## Your configuration (mandatory)
 - macOs Ventura 13.2.1
 - Python 3.10.9
 - PyMuPDF version - 1.21.1

## Additional context (optional)
This my code
```
import fitz
import json
import sys
from datetime import datetime

if len(sys.argv) != 2:
    print('Usage: python extractPDFAnnotations.py <filename>')
    sys.exit(1)
filename = sys.argv[1]

doc = fitz.open('Sandborn2003Annotated.pdf')
annotations = []

for page_num, page in enumerate(doc):
    for annot in page.annots():
        annotation_data = {}
        target_data = {}
        selector_data = {}

        # Common properties for all annotation types
        annotation_data["@context"] = [
            "https://www.w3.org/ns/anno.jsonld",
            "https://comments.acrobat.com/ns/anno.jsonld",
        ]
        annotation_data["id"] = annot.info.get("id", "")
        annotation_data["type"] = "Annotation"
        annotation_data["motivation"] = "commenting"
        annotation_data["bodyValue"] = annot.info.get("content", "")

        # Target properties
        # Replace this with the actual source identifier
        target_data["selector"] = selector_data
        annotation_data["target"] = target_data

        # Selector properties
        selector_data["node"] = {"index": page_num}
        selector_data["type"] = "AdobeAnnoSelector"

        if annot.type[0] == 8:  # Highlight annotation
            all_coordinates = annot.vertices
            highlights = []

            for i in range(0, len(all_coordinates), 4):
                quad = all_coordinates[i:i + 4]
                highlight_coord = fitz.Quad(quad).rect
                highlights.extend(
                    [highlight_coord.x0, highlight_coord.y0, highlight_coord.x1, highlight_coord.y1])

            selector_data["quadPoints"] = highlights
            # Adjust the opacity value as needed
            selector_data["opacity"] = 0.4
            selector_data["subtype"] = "highlight"
            selector_data["boundingBox"] = [
                annot.rect.x0,
                annot.rect.y0,
                annot.rect.x1,
                annot.rect.y1,
            ]
            # Adjust the stroke color as needed
            selector_data["strokeColor"] = "#fccb00"
            # Adjust the stroke width as needed
            selector_data["strokeWidth"] = 3

        # Creator properties
        annotation_data["creator"] = {
            # Replace this with the actual creator's name
            "name": annot.info.get("title", ""),
            "type": "Person",
        }

        annotations.append(annotation_data)

# Save the annotations to a JSON file
with open(filename, 'w') as f:
    json.dump(annotations, f, indent=4)

print(f'Annotations saved to {filename}')
```

**Comment by JorjMcKie** [comment]
Sorry, but I do not see where in your code you actually add the new annotation onthe new page.
Whatever, there must be an error. If doing this straightforwardly, things work as they should:
```python
old=fitz.open("Sandborn2003Annotated.pdf")
oldpage=old[0]
oldannot=oldpage.first_annot
oldannot
'Highlight' annotation on page 0 of Sandborn2003Annotated.pdf
quads = []
for i in range(0, len(oldannot.vertices), 4):
    quads.append(fitz.Quad(oldannot.vertices[i:i+4]))
new = fitz.open("Sandborn2003.pdf")
page=new[0]
page.add_highlight_annot(quads)
'Highlight' annotation on page 0 of Sandborn2003.pdf
new.save("x.pdf")
```
... delivers this as expected:
![grafik](https://user-images.githubusercontent.com/8290722/228690101-302e3afe-c34b-44fa-956f-1630991e002c.png)

**Comment by JorjMcKie** [comment]
For possible further inquiries, I am going to transfer this to the "Discussions" tab.
